### ISSUE TITLE
Android: ofGetOrientation doesn't get updated when phone is rotated

### ISSUE THREAD

**Issue by HalfdanJ** [openframeworks/openFrameworks#4508]
I have an app where the orientation is not specified (ofSetOrientation is never called) and the manifest has ```android:configChanges="orientation|screenSize"```. I can rotate the phone and the openframeworks window gets rotated, and the size gets updated. But the orientation keeps returning ```OF_ORIENTATION_DEFAULT```

Shouldn't the function return the actual orientation?

**Comment by arturoc** [comment]
yes i guess it should return the correct orientation. it's not as bad cause the orientation in OF is default 90 -90 180 where default could be read as whatever the hardware is set to even automatic rotation, and the rest are offsets from that as opposed of having orientation settings as portrait, landscape which would be absolute.

i guess having it change wouldn't break anything and would be useful though. another option would be to add a specific android function that returned the absolute rotation, as portrait, landscape...

**Comment by HalfdanJ** [comment]
How is iOS handling this?

**Comment by HalfdanJ** [comment]
@danthemellowman, @arturoc, @julapy i'm trying to make orientation work the same way on android and iOS, i've implemented an event now that triggers on orientation change, but i'm a bit unsure if the way its implemented in iOS is the way we want to go. In [the iOS example](https://github.com/openframeworks/openFrameworks/blob/master/examples/ios/iosOrientationExample/src/ofApp.mm#L195), the event triggers ofSetOrientation(). In iOS, this doesn't force the device to a certain orientation, but it does on android. So if I copy the sample, the first time you rotate the device, you freeze the orientation. 

Wouldn't it be better if ```ofGetOrientation()``` always returns the actual orientation of the device, and not what ofSetOrientation has been called with?

**Comment by danoli3** [comment]
It shouldn't force you to the device physical orientation on Android.

With iOS it doesn't force you because what if you wanted a landscape mode only App. 
Then just don't respond to the orientation commands associated with Portrait in then your own code.
I think the iOS orientation code really needs some work though.
